src/sentry_plugins/opsgenie/plugin.py

```diff
@@ -92,17 +93,15 @@ def notify_users(self, group, event, fail_silently=False, triggering_rules=None,
         if not self.is_configured(group.project):
             return
 
-        api_key = self.get_option("api_key", group.project)
-        recipients = self.get_option("recipients", group.project)
-        alert_url = self.get_option("alert_url", group.project)
-
+        client = self.get_client(group.project)
         payload = self.build_payload(group, event, triggering_rules)
-
-        headers = {"Authorization": "GenieKey " + api_key}
-
-        if recipients:
-            payload["recipients"] = recipients
-
-        resp = http.safe_urlopen(alert_url, json=payload, headers=headers)
-        if not resp.ok:
-            raise HTTPError("Unsuccessful response from OpsGenie: %s" % resp.json())
+        try:
+            client.trigger_incident(payload)
+        except Exception as e:
+            self.raise_error(e)
```

Comment:
do we know what the error object looks like? might be worth overriding the `error_message_from_json` like we did [here](https://github.com/getsentry/sentry/blob/e29cdcecabcef5b155db06b48024002f3bb2c4d3/src/sentry_plugins/pagerduty/plugin.py#L22) to provide a more useful message (depending on what the response from OpsGenie looks like)

Reply:
@MeredithAnya There's a lot of logic that goes into this function so it just depends on what the error is. It's worth noting that `raise_error` will call `error_message_from_json`. Here is an example of the error when I use the wrong API key:
```
Communicating with OpsGenie (HTTP 422): Key format is not valid!
```
I am considering making a card to use `raise_error` for all of our plugin API errors since we already use it most places.

Reply:
> It's worth noting that `raise_error` will call `error_message_from_json`.

Yeah, that's why I was suggesting to override that part since in case the useful part of the message wasn't actually saved in the 'message' attribute of the error. You may not need that here since that message looks descriptive enough